File: sort/SleepSort.py
# coding:utf-8

import time
import thread

def sort_node(number):
	time.sleep(number)
	print number
    
def sleep_sort(origin):
	for i in origin:
		thread.start_new_thread(sort_node, (i,))

sleep_sort([5, 8, 2, 3, 9, 1, 4])

File: sort/SleepSort.java
// ≈≈–ÚÀ„∑®÷ÆSleepSort

public class SleepSort {
    
    public static void main(String[] args) {
    		int[] arr = {2, 1, 5, 6, 3};
    		for (int i : arr){
    		new Thread(){
    			@Override
    			public void run(){
    				try{
    					Thread.sleep(i);
    					System.out.println(i);
    				}catch (InterruptedException e) {
						e.printStackTrace();
					}
    			}
    		}.start();	
    		}
      }
}